Sources/ForageSDK/Component/ForagePINTextField/Vault/BtPINTextField.swift

```diff
@@ -95,6 +97,22 @@ class BasisTheoryTextFieldWrapper: UIView, VaultWrapper {
                 self._isValid = elementEvent.valid
                 self._isComplete = elementEvent.complete
                 self.delegate?.textFieldDidChange(self)
+            } else if (
+                elementEvent.type == BTEventType.FOCUS.rawValue ||
+                elementEvent.type == BTEventType.BLUR.rawValue
+            ) {
+                // Note: This is custom logic to unblock us while BT makes changes on their
+                // end. They had the incorrect initial state logic for their input field.
+                if (elementEvent.empty) {
+                    self._isEmpty = elementEvent.empty
+                    self._isValid = false
+                    self._isComplete = false
+                } else {
+                    self._isEmpty = elementEvent.empty
+                    self._isValid = elementEvent.valid
+                    self._isComplete = elementEvent.complete
+                }
+                self.delegate?.textFieldDidChange(self)
```

Comment:
Let me know if this is my lack of iOS understanding here but I wanted to clarify if we should be calling `self.delegate?.textFieldDidChange(self)` here or `self.delegate?.focusDidChange(self)`? It seems like this statement is living withing:
```swift
else if (
    elementEvent.type == BTEventType.FOCUS.rawValue ||
    elementEvent.type == BTEventType.BLUR.rawValue
) {
  // ...
  self.delegate?.textFieldDidChange(self) // should this be self.delegate?.focusDidChange(self) ?

}
```

Reply:
Yikes you're right!

Reply:
Ah OK - glad I wasn't going crazy there! I saw that you fixed this in [`fdbadf8`](https://github.com/teamforage/forage-ios-sdk/pull/72/commits/fdbadf83f75b8da6e698d9f084da27595e764221). Will re-view now!